src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -491,6 +491,11 @@ internal override IOperation CloneOperationCore(IOperation operation)
         /// </summary>
         public abstract SymbolInfo GetSymbolInfo(SelectOrGroupClauseSyntax node, CancellationToken cancellationToken = default(CancellationToken));
 
+        /// <summary>
+        /// Gets the semantic information associated with a LINQ query syntax clause.
+        /// </summary>
+        public abstract SymbolInfo GetSymbolInfo(QueryClauseSyntax node, CancellationToken cancellationToken = default(CancellationToken));
+
```

Comment:
It is not clear why this is needed, we have existing API that return information about QueryClauseSyntax, for example GetQueryClauseInfo.